tagging/api.py

```diff
@@ -86,27 +96,28 @@ def get_datasets(
     offset: Optional[int] = FastQuery(0, alias="page[offset]"),
     limit: Optional[int] = FastQuery(DEFAULT_PAGE_SIZE, alias="page[limit]")
 
-) -> List[Dataset]:
+) -> List[PersistedDataset]:
 
 
     """ Searches datasets based on query parameters. Provides pagine through skip and limit
 
     Args:
         uri (Optional[str], optional): find dataset based on uri. Defaults to None.
-        tags(Optional[List[str]], optional): lsit of tags to search for. Defaults to none.
+        tags(Optional[List[str]], optional): list of tags to search for. Defaults to none.
         skip (Optional[int], optional): [description]. Defaults to 0.
         limit (Optional[int], optional): [description]. Defaults to 10.
 
     Returns:
-        List[Dataset]: [Full object datasets corresponding to search prarmeters]
+        List[PersistedDataset]: [Full object datasets corresponding to search parameters]
     """
     return svc_context.tag_svc.find_datasets(offset=offset, limit=limit, uri=uri, tags=tags)
 
 
-@app.patch(API_URL_PREFIX + '/datasets/{uid}/tags', tags=['datasets', 'tags'], response_model=CreateResponseModel)
-def add_tags(uid: str, tags: List[Tag]):
-    new_asset = svc_context.tag_svc.add_tags(tags, uid)
-    return CreateResponseModel(uid=new_asset.uid)
+@app.patch(API_URL_PREFIX + '/datasets/{uid}/tags', tags=['datasets', 'tags'], response_model=CreateTagPatchResponse)
+def modify_tags(uid: str, req: TagPatchRequest):
+    added_tags_uid = svc_context.tag_svc.add_tags(req.add_tags, uid)
```

Comment:
It would be more efficient if this could be accomplished with one call to tag_svc...especially since each call involves a lookup of the same Dataset collection.